network/p2p/p2pnode/libp2pNode.go

```diff
@@ -532,6 +532,7 @@ func (n *Node) IsConnected(peerID peer.ID) (bool, error) {
 // SetRouting may be called at most once.
 func (n *Node) SetRouting(r routing.Routing) {
 	if n.routing != nil {
+		// we should not allow overriding the routing implementation if one is already set; crashing the node.
 		n.logger.Fatal().Msg("routing already set")
```

Comment:
Wondering if we should start returning the error to the caller and allow the error to be thrown by the parents context?

Reply:
https://github.com/onflow/flow-go/pull/4800/commits/9311c0c466e1f201806f1c06f5bd034fcbb738cd